src/generate/renders/extraExt/CliExtReadme.ts

```diff
@@ -100,8 +43,66 @@ export class CliExtReadme extends TemplateBase {
         }
     }
 
-    public async GetRenderData(model: CodeModelAz): Promise<string[]> {
-        const output: string[] = [];
-        return output;
+    public async GetRenderData(model: CodeModelAz): Promise<Record<string, any>> {
+        const { exampleHandler } = this.model.GetHandler();
+
+        const exampleConverter = (item: any) => {
+            item.title = item.Method.charAt(0).toUpperCase() + item.Method.slice(1);
+            item.CommandStringLines = CmdToMultiLines(item.CommandString);
+            const waitCommandString = exampleHandler.GetExampleWait(item).join(' ');
+            if (!isNullOrUndefined(waitCommandString) && waitCommandString !== '') {
+                item.CommandStringLines.push('');
+                const temp = CmdToMultiLines(waitCommandString);
+                item.CommandStringLines.push(...temp);
+            }
+            return {
+                title: item.title,
+                CommandStringLines: item.CommandStringLines,
+                HttpMethod: item.HttpMethod,
+            };
+        };
+
+        const commandConverter = (item: any) => {
+            item.examples = exampleHandler.GetExamples(false);
+            for (let i = 0; i < item.examples.length; i++) {
+                item.examples[i] = exampleConverter(item.examples[i]);
+            }
+            return item;
+        };
+
+        const inputProperties: Map<CodeModelTypes, RenderInput> = new Map<
+            CodeModelTypes,
+            RenderInput
+        >([
+            ['extension', new RenderInput(['name'])],
+            ['commandGroup', new RenderInput(['name'])],
+            ['command', new RenderInput([], {}, [], commandConverter)],
+        ]);
+
+        const dependencies = <[CodeModelTypes, CodeModelTypes][]>[
+            ['extension', 'commandGroup'],
+            ['commandGroup', 'command'],
+        ];
+        const data = model.getModelData('extension', inputProperties, dependencies);
+        for (const extension of data.Extensions) {
+            for (const group of extension.CommandGroups) {
+                group.examples = [];
+                for (const command of group.Commands) {
+                    group.examples = group.examples.concat(command.examples);
+                }
+                group.Commands = undefined;
+
+                for (let i = 0; i < group.examples.length - 1; ++i) {
```

Comment:
Though this is the existing code, but is it possible to optimize the loop?

Reply:
@changlong this is actually a bubble sort logic. we can replace it with something like array.sort()

Reply:
Hi @lirenhe @qiaozha ,

Changing to a common sorting algorithm will bring breaking change in the result (since the original algorithm is non-stable sorting though the result is stable for same input). And the javascript array.sort() has different result across nodejs version (for those items who are 'equal with' each other), so I tend to keep current sorting loop.

In commit d44cd922da8ccf9cb346196d582bd0f0b371aa61, I made a brief 'swap' logic to make the code clean.
How do you think about it?

Reply:
ok. considering the current stage, it's better not to introduce more breaking changes now.